src/components/Drawer/Drawer.tsx

```diff
@@ -0,0 +1,236 @@
+   * If true, the overlay will not be rendered, scrolling for the entire page will remain enabled,
+   * and focus will not be locked to the contents of the drawer
+   */
+  hideOverlay?: boolean;
+  /**
+   * By default the first focusable element will receive focus when the dialog
+   * opens but you can provide a ref to focus instead.
+   *
+   * @see Docs https://reach.tech/dialog#dialog-initialfocusref
+   */
+  initialFocusRef?: RefObject<HTMLDivElement>;
+  /**
+   * If the drawer is open
+   */
+  isOpen: boolean;
+  /**
+   * Which edge of the viewport should the drawer appear from
+   */
+  placement?: DrawerPlacementType;
+  /**
+   * Function that is called whenever the user either hits
+   *  the "Escape" key, clicks the close button icon, or clicks the overlay.
+   */
+  onDismiss?: (event?: React.SyntheticEvent) => void;
+  /**
+   * Title to be displayed at the top of the Drawer.
+   * A close button will be rendered automatically if this prop is defined.
+   */
+  title?: string;
+  /**
+   * The width of the Drawer when opened. Can be given a standard css value (px, rem, em, %),
+   * or a [width token](/?path=/story/design-tokens-design-tokens--page#width)
+   */
+  width?: DimensionSize | string;
+}
+
+export const Drawer: FC<DrawerProps> = forwardRef<HTMLDivElement, DrawerProps>(
+  (
+    {
+      ariaLabel,
+      ariaLabelledBy,
+      allowPinchZoom = false,
+      children,
+      className,
+      closeButton = false,
+      closeOnOverlayClick = true,
+      containerRef = undefined,
+      dangerouslyBypassFocusLock = false,
+      dangerouslyBypassScrollLock = false,
+      hideOverlay = false,
+      initialFocusRef,
+      isOpen,
+      onDismiss,
+      placement = 'right',
+      title,
+      width = undefined,
+    },
+    ref,
+  ) => {
+    const dynamicWidth = (WIDTH_OPTIONS as Array<string | undefined>).includes(width)
+      ? `var(--size-width-${width})`
+      : width;
+
+    const dynamicStyle: CSSProperties = {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      ['--w' as any]: dynamicWidth,
+    };
+
+    const overlayClassnames = classNames(styles.overlay, styles.drawer, {
+      [styles['hide-overlay']]: hideOverlay,
+      [styles['hide-overlay-left']]: hideOverlay && placement === 'left',
+      [styles['hide-overlay-right']]: hideOverlay && placement === 'right',
+      [styles['hide-overlay-bottom']]: hideOverlay && placement === 'bottom',
+      [styles['hide-overlay-top']]: hideOverlay && placement === 'top',
+      'position-fixed': containerRef === undefined,
+      'position-absolute': containerRef !== undefined,
+    });
+
+    const contentClassnames = classNames(styles['drawer-content'], {
+      [styles.bottom]: placement === 'bottom',
+      [styles.left]: placement === 'left',
+      [styles.right]: placement === 'right',
+      [styles.top]: placement === 'top',
+      [styles['hide-overlay']]: hideOverlay,
+      'overflow-auto': !closeButton && !title,
```

Comment:
If the drawer has a header, don't apply scrolling to the entire content div so the header can be fixed.